sf-recco/ios-sdk
# SHAD-459 Support questionaries as new reccomendations

https://vilua.atlassian.net/browse/SHAD-459

## What 

Android equivalent (https://github.com/SignificoHealth/recco-android-sdk/pull/73). 

- Supports a new type of content to be shown in the dashboard: 'questionnaires`.
- These are shown when received from backend in the tailored sections.
- It gets rid of the 'Load' views that were shown when content was partially locked. 
- Shows questionnaires cards instead. Navigates to them when clicking.

## How 

- Brings the latest changes from the Open Api schema. 
- Requests a new content type to backend for the tailored recommendations. 
- Shows a item view per questionnaire in sections.
- Assumes that `.partiallyUnlocked` is now `.unlocked`.

## How to test

1. Log with any account.
2. Pass the onboarding to the dashboard.
3. Scroll to a locked content.
4. Unlock it **by not answering questions** just tapping next.
5. Notice the new questionnaire cards.
6. Tap one, should take you to the questionnaire.
7. Complete it. Card should now dissapear from the section.

## Screenshots

| Light | Dark |
| -- | -- |
| <img height="500" src="https://github.com/SignificoHealth/recco-ios-sdk/assets/3531999/96fe29b7-6000-4a03-95b2-4690acd4009d" /> | <img height="500" src="https://github.com/SignificoHealth/recco-ios-sdk/assets/3531999/620c4246-bcd2-4a64-a140-1f0cf16602d3"> | 

| Light | Dark |
| -- | -- |
| <img height="500" alt="Screenshot 2023-11-29 at 15 50 21" src="https://github.com/SignificoHealth/recco-ios-sdk/assets/3531999/902998ab-ec99-4381-b40d-d6d4355d9e94"> |  <img height="500" alt="Screenshot 2023-11-29 at 15 50 28" src="https://github.com/SignificoHealth/recco-ios-sdk/assets/3531999/39eadab2-c1ef-4b44-a1c8-e4959c8729e4"> |
